src/node.cc

```diff
@@ -2628,6 +2633,17 @@ static void EnvGetter(Local<Name> property,
 static void EnvSetter(Local<Name> property,
                       Local<Value> value,
                       const PropertyCallbackInfo<Value>& info) {
+  if (config_pending_deprecation && !env_nonstring_deprecation_warned &&
+      !value->IsString()) {
+    if (ProcessEmitDeprecationWarning(
+          Environment::GetCurrent(info),
+          "Assigning any value other than a string to a process.env property "
+          "is deprecated. Please make sure to convert the value to a string "
+          "before setting process.env with it.",
+          "DEP01XX").IsNothing())
+      return;
+    env_nonstring_deprecation_warned = true;
+  }
```

Comment:
Thank you for this, I would expect it to become pretty helpful when trying to figure out actual ecosystem usage.